envoy/upstream/upstream.h

```diff
@@ -1146,7 +1146,7 @@ class ClusterInfo : public Http::FilterChainFactory {
   /**
    * @return eds cluster service_name of the cluster.
    */
-  virtual absl::optional<std::string> edsServiceName() const PURE;
```

Comment:
This field is optional in the proto API. And this optionality is expressed by the empty value. So for C++ API returning `std::optional<std::string>` or just empty `std::string` should be equivalent.

Given this, I would go with returning `absl::string_view` and making it empty when `service_name` was not set in the API. This should remove the need for the optional wrapper.

Reply:
if the empty checking could expresent `it's was not set`, IMO, it's better. I will update it.

Reply:
Although I prefer string view, it seems that the `const std::string&` is more appropriate here. Because lots of pb setting cannot accept the string view.